test/cpp/test/mock_stream_test.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ * Copyright 2020 the gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpcpp/test/mock_stream.h>
+#include <gtest/gtest.h>
+
+#include "absl/memory/memory.h"
+
+#include "src/proto/grpc/testing/echo.grpc.pb.h"
+
+using grpc::testing::EchoRequest;
+using grpc::testing::EchoResponse;
+
+TEST(MockStreamTest, Basic) {
+  auto cr = absl::make_unique<grpc::testing::MockClientReader<EchoResponse>>();
+  ASSERT_NE(cr, nullptr);
```

Comment:
Are these meaningful checks? Can absl::make_unique ever return nullptr? Is there some other  check that would make this more useful of a test?

Reply:
Yeah, this is basically to make sure that they're not abstract classes. (meaning all methods are mocked) I guess this should be enough.

Reply:
Would a more useful test be then to actually derive mocks from these and then use them the way that a user actually would? I'm not fond of the idea of having a test that's basically just a build.

Reply:
I don't think it gives more test coverage, though because it's testing the mock feature of gtest. If the class is ready to be mockable, I think it's enough for this test? This test is designed to make sure those classes are mockable. (Previously those were not)